tests/test_modeling_common.py

```diff
@@ -4525,65 +4359,6 @@ def test_sdpa_matches_eager_sliding_window(self):
                     torch.allclose(res_eager[attention_mask == 1], res_sdpa[attention_mask == 1], rtol=1e-4, atol=1e-4)
                 )
 
-    @require_flash_attn
-    @require_torch_gpu
-    @mark.flash_attn_test
-    @slow
-    def test_flash_attn_2_generate_use_cache(self):
```

Comment:
This test was pretty toothless -- it didn't even check the actual output of `generate`.

I've added `test_eager_matches_fa2_generate` in its place, which checks the outputs. It's a clone of the SDPA generate equivalence test, except that it uses FA2.